### DISCUSSION TITLE
Format a list with dedicated wrapping characters

### DISCUSSION THREAD

**Discussion by iliashkolyar** [axuno/SmartFormat#321] [Q&A] [answered]
Hello, 
It seems like a fairly easy thing to do, but I couldn't find the way to do it ([from the following wiki](https://github.com/axuno/SmartFormat/wiki/Lists-_-ListFormatter)).

I would like to format a list of strings while wrapping each entry with some character.
So let's say the character is `'*'`, I would like the following outputs for each input:
```
var input = new List<string> { "one", "two" };
// Output = "*one*, *two*"

var input = new List<string>();
// Output = ""
```

Thanks in advance!

**Comment by axunonb** [comment, marked as answer]
The members of the list are prepended with a `*` and appended with a `*`. The separator is ` ,`.
The empty list should give an empty result.
So you could e.g. go for this:
```CSharp
var result = Smart.Format("{0.Count:choose(0):|{0:list:*{}*|, }}", input);
//
// for input = new List<string> { "one", "two" };
// output: "*one*, *two*"
//
// for input = new List<string>();
// output: ""
```

**Comment by iliashkolyar** [comment]
First of all, thanks a lot for your prompt answer!

This really works well for the latest version, but unfortunately i'm using version `2.5.3` in which the `:*{}*` doesn't seem to do anything.

I've tried upgrading the version to the latest, but it broke the build since I use the (now deprecated) `Smart.Default.Parser.UseAlternativeBraces` API.
I'm using it to be able to format JSON payloads, so I needed to use different braces than `{` and `}`.

Could you please suggest what is the way of using "alternative braces" in the newest version?
Thanks again!

**Comment by axunonb** [comment]
Try setting `SmartSettings.StringFormatCompatibility = true`. This has some more implications as described in the Wiki, but might work in your case. The best way, however, is to keep `SmartSettings.StringFormatCompatibility = false` and to escape the JSON curly brace literals in the format string. Alternate braces were I nightmare to maintain - too many side effects.
